serverless/serverless
# Fixed a bug when using dot notation in YAML keys

## What did you implement:

Closes #3607

Fixed a bug when using dot notation in YAML keys. The variable system didn't handle that edge case and converts the dot notation to nested objects.



## How did you implement it:



The issue was caused by the deepMapValues module/mixin for lodash. It didn't handle that edge case so the property path returned was not reliable. I fixed it by removing dependency on that module and taking/customizing the logic and maintain it within the codebase. The edit mainly was changing the representation of property paths to be with arrays instead of strings.

## How can we verify it:
package the following config and check the outputs. You should find the `some.dots` property properly set.

```yml
service: vars-in-files

provider:
  name: aws
  runtime: nodejs6.10

custom:
  dots:
    some.dots: hello

functions:
  hello:
    handler: handler.hello

resources:
  Outputs:
    varA:
      Value: ${self:custom.dots}
      Export:
        Name: varA
```


## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
